portality/tasks/check_latest_es_backup.py

```diff
@@ -0,0 +1,72 @@
+from portality import models
+from portality.core import app
+
+from portality.lib.es_snapshots import ESSnapshotsClient
+
+from portality.tasks.redis_huey import main_queue, schedule
+from portality.decorators import write_required
+
+from portality.background import BackgroundTask, BackgroundApi
+
+
+class CheckLatestESBackupBackgroundTask(BackgroundTask):
+
+    __action__ = "check_latest_es_backup"
+
+    def run(self):
+        """
+        Execute the task as specified by the background_job
+        :return:
+        """
+        client = ESSnapshotsClient()
+        client.check_today_snapshot()
+
+    def cleanup(self):
+        """
+        Cleanup after a successful OR failed run of the task
+        :return:
+        """
+        pass
+
+    @classmethod
+    def prepare(cls, username, **kwargs):
+        """
+        Take an arbitrary set of keyword arguments and return an instance of a BackgroundJob,
+        or fail with a suitable exception
+
+        :param kwargs: arbitrary keyword arguments pertaining to this task type
+        :return: a BackgroundJob instance representing this task
+        """
+
+        # first prepare a job record
+        job = models.BackgroundJob()
+        job.user = username
+        job.action = cls.__action__
+        return job
+
+    @classmethod
+    def submit(cls, background_job):
+        """
+        Submit the specified BackgroundJob to the background queue
+
+        :param background_job: the BackgroundJob instance
+        :return:
+        """
+        background_job.save()
+        check_latest_es_backup.schedule(args=(background_job.id,), delay=10)
+
+
+@main_queue.periodic_task(schedule("check_latest_es_backup"))
```

Comment:
I see you have this on the main_queue and the other task on the long_running queue.  This seems sensible, and I just wanted to double check that this is correct, and that this checking task should be on the main_queue and not the long_running queue.  (I don't know how long it will take to run, it might be super-quick)

Reply:
I did intentionally move the deletion to the long running queue, but I think I may want the check to go there too. Right now, it takes 2-5 minutes to complete because there's so much metadata to return. When the prune is ongoing, it'll be quicker. I'll move it though.

Reply:
Actually, we definitely want to run the check every day, and the main queue is the right place for that. I'll leave it in the main queue.